diff --git a/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/BUILD.bazel b/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/BUILD.bazel
index 330e38d..1fc8f36 100644
--- a/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/BUILD.bazel
+++ b/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/BUILD.bazel
@@ -2,7 +2,13 @@ load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
 go_library(
     name = "ffi",
-    srcs = ["ffi.go"],
+    srcs = [
+        "ffi.go",
+        "//internal/ffi/native:polar.h",  # keep
+    ],
+    cdeps = [
+        "//internal/ffi/native/linux:libpolar",  # keep
+    ],
     cgo = True,
     clinkopts = select({
         "@io_bazel_rules_go//go/platform:android_amd64": [
@@ -21,7 +27,7 @@ go_library(
             "internal/ffi/native/macos/arm64/libpolar.a -ldl -lm",
         ],
         "@io_bazel_rules_go//go/platform:linux_amd64": [
-            "internal/ffi/native/linux/libpolar.a -ldl -lm",
+            "external/com_github_osohq_go_oso/internal/ffi/native/linux/libpolar.a -ldl -lm",  # keep
         ],
         "@io_bazel_rules_go//go/platform:windows_amd64": [
             "internal/ffi/native/windows/libpolar.a -lm -lws2_32 -luserenv -lbcrypt",
diff --git a/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/native/BUILD.bazel b/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/native/BUILD.bazel
index f08b997..99788be 100644
--- a/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/native/BUILD.bazel
+++ b/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/native/BUILD.bazel
@@ -20,3 +20,7 @@ alias(
     actual = ":native",
     visibility = ["//:__subpackages__"],
 )
+
+exports_files([
+    "polar.h",
+])
diff --git a/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/native/linux/BUILD.bazel b/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/native/linux/BUILD.bazel
index 3c4d8ec..5a54f5d 100644
--- a/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/native/linux/BUILD.bazel
+++ b/bazel-oso/app/third_party/com_github_osohq_go_oso/internal/ffi/native/linux/BUILD.bazel
@@ -1,4 +1,5 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@rules_cc//cc:defs.bzl", "cc_import")
 
 go_library(
     name = "linux",
@@ -12,3 +13,12 @@ alias(
     actual = ":linux",
     visibility = ["//:__subpackages__"],
 )
+
+cc_import(
+    name = "libpolar",
+    hdrs = [
+        "//internal/ffi/native:polar.h",
+    ],
+    static_library = "libpolar.a",
+    visibility = ["//visibility:public"],
+)
